src/niclips/image/_convert.py

```diff
@@ -62,53 +59,49 @@ def normalize(
     vmax: float | None = None,
 ) -> np.ndarray:
     """Normalize data using vmin/vmax if provided, otherwise data min/max."""
-    data = np.asarray(data)
-    if vmin is None:
-        vmin = np.nanmin(data)
-    if vmax is None:
-        vmax = np.nanmax(data)
-    data = (data - vmin) / max(vmax - vmin, EPS)
-    data = np.clip(data, 0, 1)
-    return data
+    vmin = np.nanmin(data) if vmin is None else vmin
+    vmax = np.nanmax(data) if vmax is None else vmax
+    scale = vmax - vmin
+    if scale > EPS:
+        return np.clip((data - vmin) / scale, 0, 1)
+    return np.zeros_like(data)
 
 
 def scale(
     img: Image.Image, height: int, resample: Image.Resampling | None = None
 ) -> Image.Image:
-    """Scale an image to a target height.
+    """Scale image isotropically to a target height.
 
-    Ensure width is even numbered.
+    Note: Ensure size is even numbered for video codec.
     """
-    height += height % 2
-    scale = height / img.height
-    width = int(scale * img.width) + (int(scale * img.width) % 2)
-    size = width, height
-    img = img.resize(size, resample=resample)
-    return img
+    # Currently rescale to isotropic to not stretch images in generated figures
+    # Can maintain aspect ratio by using image size (width, height)
+    # Leaving the code below to do so.
+    # scale = height / img.height
+    # width = int(img.width * scale) & ~1
+    height = height & ~1
+
+    return img.resize((height, height), resample=resample)
```

Comment:
Seems like a possible typo here? What happened to the width?

Also the bitwise operation `height = height & ~1` is a bit obscure.

Reply:
> Seems like a possible typo here? What happened to the width?

So this wasn't a typo 😅 - I left the width code commented out there in case we wanted to use it but wanted to discuss it. In the original code, we resampled 3x to enforce the image is isotropic (which was also part of the slowness before). With `(height, height)` here, we just resample once to make the output isotropic at the very end for the frame. We could also do `(width, height)`, which will retain the original voxel size. 

> Also the bitwise operation height = height & ~1 is a bit obscure.

I agree about the bitwise operation being obscure, but it did make a notable improvement in the time it takes for ensuring the dimension is even numbered (versus the modulo approach earlier). I'll add a more clear comment to this.

Reply:
Added additional comments regarding the bitwise operation in 1fee355.

See response to other comment below regarding scaling + resampling.